src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/ApplicationServiceTest.kt

```diff
@@ -1646,6 +1690,178 @@ class ApplicationServiceTest {
+          username,
+          "jwt",
+          user.probationRegion.id,
+        )
+
+      result as AuthorisableActionResult.Success
+      val validatableActionResult = result.entity as ValidatableActionResult.Success
+      val persistedApplication = validatableActionResult.entity as ApprovedPremisesApplicationEntity
+
+      assertThat(persistedApplication.applicantUserDetails).isEqualTo(theUpdatedApplicantUserDetailsEntity)
+      assertThat(persistedApplication.caseManagerIsNotApplicant).isEqualTo(true)
+      assertThat(persistedApplication.caseManagerUserDetails).isEqualTo(theUpdatedCaseManagerUserDetailsEntity)
+    }
+
+    @Test
+    fun `updateApprovedPremisesApplication if applicant is now case manager, removes existing case manager user details`() {
+      submitApprovedPremisesApplication = SubmitApprovedPremisesApplication(
+        translatedDocument = {},
+        isPipeApplication = true,
+        isWomensApplication = false,
+        isEmergencyApplication = false,
+        isEsapApplication = false,
+        targetLocation = "SW1A 1AA",
+        releaseType = ReleaseTypeOption.licence,
+        type = "CAS1",
+        sentenceType = SentenceTypeOption.nonStatutory,
+        situation = SituationOption.bailSentence,
+        applicantUserDetails = Cas1ApplicationUserDetails("applicantName", "applicantEmail", "applicantPhone"),
+        caseManagerIsNotApplicant = false,
+        caseManagerUserDetails = null,
+      )
+
+      val application = ApprovedPremisesApplicationEntityFactory()
+        .withApplicationSchema(newestSchema)
+        .withId(applicationId)
+        .withCreatedByUser(user)
+        .withSubmittedAt(null)
+        .withApplicantUserDetails(Cas1ApplicationUserDetailsEntity(UUID.randomUUID(), "applicantName", "applicantEmail", "applicantPhone"))
+        .withCaseManagerUserDetails(Cas1ApplicationUserDetailsEntity(UUID.randomUUID(), "oldCaseManEmail", "oldCaseManName", "oldCaseManPhone"))
+        .produce()
+        .apply {
+          schemaUpToDate = true
+        }
+
+      setupMocksForSuccess(application)
+
+      val existingApplicantUserDetails = application.applicantUserDetails!!
+      val existingCaseManagerUserDetails = application.caseManagerUserDetails!!
+
+      every {
+        mockCas1ApplicationUserDetailsRepository.save(match { it.id == existingApplicantUserDetails.id })
+      } answers { it.invocation.args[0] as Cas1ApplicationUserDetailsEntity }
+
+      every { mockCas1ApplicationUserDetailsRepository.delete(existingCaseManagerUserDetails) } returns Unit
+
+      val result = applicationService.submitApprovedPremisesApplication(
+        applicationId,
+        submitApprovedPremisesApplication,
+        username,
+        "jwt",
+        user.probationRegion.id,
+      )
+
+      assertThat(result is AuthorisableActionResult.Success).isTrue
+
+      verify { mockCas1ApplicationUserDetailsRepository.delete(existingCaseManagerUserDetails) }
```

Comment:
Do we also want an integration test to verify that the API now exposes `caseManagerUserDetails:  null` when the `caseManagerIsNotApplicant` flag is toggled to false? Or is it sufficient to know that the service asks the repository to delete the `Cas1ApplicationUserDetailsEntity` record?

Reply:
I think unit test is sufficient in this case